### PR TITLE
test: add tests for user settings DET-6788 DET-6750

### PR DESCRIPTION
**Pull request by johnkim-det**
## Description

Includes tests for `useModalUserSettings`, `useModalChangeName`, `useModalChangePassword`, `Avatar`, and `utils/user`. Also includes some minor refactoring.

## Test Plan
- [ ] make sure test suite runs without errors

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/hooks/useModal/UserSettings/useModalChangeName.test.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+import StoreProvider, { StoreAction, useStoreDispatch } from 'contexts/Store';
+import { PatchUserParams } from 'services/types';
+import { DetailedUser } from 'types';
+
+import useModalUserSettings from './useModalUserSettings';
+
+jest.mock('services/api', () => ({
+  patchUser: ({ userParams }: PatchUserParams) => {
+    return Promise.resolve({ displayName: userParams.displayName });
```
Comment (EmilyBonar) [inline webui/react/src/hooks/useModal/UserSettings/useModalChangeName.test.tsx:14]: nit: this should return the whole user object, not just the display name. I think it also needs to update the value in the store.

File: webui/react/src/hooks/useModal/UserSettings/useModalUserSettings.test.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+  return (
+    <div>
+      {contextHolder}
+      <Button onClick={() => openUserSettingsModal()}>
+        {OPEN_MODAL_TEXT}
+      </Button>
+      <Button onClick={() => loadUsers()}>
+        {LOAD_USERS_TEXT}
+      </Button>
```
Comment (EmilyBonar) [inline webui/react/src/hooks/useModal/UserSettings/useModalUserSettings.test.tsx:53]: question: once again, why is this a button?

File: webui/react/src/hooks/useModal/UserSettings/useModalUserSettings.test.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    expect(screen.getByText(DISPLAY_NAME)).toBeInTheDocument();
+    expect(screen.getByText(USERNAME)).toBeInTheDocument();
+    expect(screen.getByText(CHANGE_NAME_TEXT)).toBeInTheDocument();
+    expect(screen.getByText(CHANGE_PASSWORD_TEXT)).toBeInTheDocument();
+  });
+
+  it('closes the modal', async () => {
+    await setup();
+
+    userEvent.click(screen.getByRole('button', { name: /cancel/i }));
```
Comment (EmilyBonar) [inline webui/react/src/hooks/useModal/UserSettings/useModalUserSettings.test.tsx:82]: chore: this is clicking the cancel button in the footer, which is hidden to users. instead, it should be clicking the X button in the top right corner

**Review by EmilyBonar — APPROVED** [review]
This is basically good, just address the comments and make sure the tests are working in CI

File: webui/react/src/components/Avatar.test.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  );
+};
+
+const setup = () => {
+  render(
+    <StoreProvider>
+      <TestApp />
+    </StoreProvider>,
+  );
+};
```
[the remark this answers is not in the document]
Comment (johnkim-det) [reply webui/react/src/components/Avatar.test.tsx:55]: Avatar functionality has changed to display differently depending on the user object it looks up in the store

**Pull request merged**